merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/MissionModelProcessor.java

```diff
@@ -80,22 +82,48 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
     for (final var element : roundEnv.getElementsAnnotatedWith(MissionModel.class)) {
       final var packageElement = (PackageElement) element;
       try {
-        final var missionModelRecord = missionModelParser.parseMissionModel(packageElement);
+        final var missionModelRecord$ = missionModelParser.parseMissionModel(packageElement);
 
         final var generatedFiles = new ArrayList<>(List.of(
-            missionModelGen.generateMerlinPlugin(missionModelRecord),
-            missionModelGen.generateSchedulerPlugin(missionModelRecord)));
+            missionModelGen.generateMerlinPlugin(missionModelRecord$),
+            missionModelGen.generateSchedulerPlugin(missionModelRecord$)));
 
-        missionModelRecord.modelConfigurationType
-            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord, configType))
+        missionModelRecord$.modelConfigurationType
+            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord$, configType))
             .ifPresent(generatedFiles::add);
 
         generatedFiles.addAll(List.of(
-            missionModelGen.generateMissionModelFactory(missionModelRecord),
-            missionModelGen.generateSchedulerModel(missionModelRecord),
-            missionModelGen.generateActivityActions(missionModelRecord),
-            missionModelGen.generateActivityTypes(missionModelRecord)
+            missionModelGen.generateMissionModelFactory(missionModelRecord$),
+            missionModelGen.generateSchedulerModel(missionModelRecord$),
+            missionModelGen.generateActivityActions(missionModelRecord$),
+            missionModelGen.generateActivityTypes(missionModelRecord$)
         ));
+
+        final var autoValueMapperRequests = roundEnv.getElementsAnnotatedWith(AutoValueMapper.class).stream().map(Element::asType).toList();
+        {
+          final var recordType = elementUtils.getTypeElement("java.lang.Record").asType();
+          for (final var request : autoValueMapperRequests) {
+            if (!typeUtils.isSubtype(request, recordType)) {
+              messager.printError("AutoValueMapper is only allowed on record types. %s is not a record".formatted(request.toString()));
+            }
+          }
+        }
+        final var autoValueMappers = missionModelGen.generateAutoValueMappers(
+            missionModelRecord$,
+            autoValueMapperRequests);
+        generatedFiles.add(autoValueMappers.getLeft());
```

Comment:
**TL;DR**: We should only process `@AutoValueMapper` annotations observed on the return types of activity effect models, not all such annotations. If we want to extend `@AutoValueMapper` to non-activity-return cases, we can deploy another scoping mechanism (or lean on an existing one). While `@AutoValueMapper` should not be "owned" by activity type processing, it's not a wholly independent process, either.

Looking ahead to a better-behaved processor, this approach of generating a single mapper family for all `@AutoValueMapper`s observed in the first round (and thus ignoring / failing on later rounds) probably isn't tenable in the long term. I don't think we need to change that behavior but we *should* think about what kind of annotation design will allow us to do better in the future with a minimum of churn for customers.

As I understand it, the value of `@AutoValueMapper` comes in two parts:
* The generation of a `ValueMapper` implementation for each annotated type.
* The generation of a mapper family with a factory method for each `ValueMapper` generated above.

The first part is easy to do incrementally, since each mapper implementation is independent of the others.

The second is not so easy, because we can only generate the mapper family once we know about all of the mappers it needs to reference. If we just go off of the list returned by `roundEnv.getElementsAnnotatedWith()` in one round, then we will miss any `@AutoValueMapper`-annotated elements generated in later rounds. There are basically three ways to handle this kind of aggregate output:

1. Don't output the mapper family until the final round. This lets us accumulate all `@AutoValueMapper` requests until the last possible moment, but other annotation processors won't be able to process the resulting mapper family. This may be an acceptable solution for some kinds of output that aren't meant to be processed further.
2. Limit the scope of `@AutoValueMapper`s considered, so that we can enumerate a full list early in the process and thus know exactly when there are no further mappers to consider. This is exactly what `@WithActivityType` does for activity types; we provide a courtesy check that all `@ActivityType`-annotated elements are claimed by a `@WithActivityType`, but morally, we're only processing activity types referenced from the `package-info.java` annotations. We can do this explicitly (e.g. `@WithAutoMapperFor(...)`) or implicitly ("all and only `@AutoValueMapper`s on return values from an activity type's effect model), but we must ultimately circumscribe some closed family of elements that we actually process.
3. For each `@AutoValueMapper`, generate a mapper family with one factory method, rather than accumulating all of them into a single family. We still need some way to know when we have a full set of `TypeRule`s, which implies that a user would have to add an `@WithMappers` for every single output type.

You might think we could get away with (1) here, because we don't actually need to process the generated mapper family since we're generating its `TypeRule`s ahead of time. Unfortunately, we have a similar problem, in that we don't want to resolve mappers for types until we have all of the `TypeRule`s available. If we started resolving without all of the rules, we might end up using an implementation we wouldn't have. As a very concrete example, we currently have a catch-all rule for any enum. If an activity returns an enum annotated with `@AutoValueMapper` (ignoring that it's currently built only for records, I hope we see that this is a splinter of the full story of use), then we might accidentally use the catch-all rather than the specific enum mapper. What a hard problem to debug!

I also think (3) is a non-starter, because it basically requires a user to add both a `@WithMappers` and a `@WithActivityType` for each activity type. This seems redundant, and also easy to forget about when developing an activity type (e.g. changing its return from `void` to an actual output).

On the other hand, I think (2) is pretty tenable. `@AutoValueMapper` would not be processed in the absence of other information (because we run into the open family issue again), but rather describes an inert desire *in the scope of* some other process. We can generate an `OutputAttributesMappers` class covering all *and only* the value mappers for the processed activity types, which is a closed family because the activity types are themselves a closed family (enumerated by `@WithActivityTypes`.

Reply:
I've updated the code to only generate value mappers for records that are the return type of an effect model.